From e9e6d5ded4c3e53254438a73a79f04209035603e Mon Sep 17 00:00:00 2001
From: Mathieu Poirier <mathieu.poirier@linaro.org>
Date: Mon, 4 Mar 2024 13:38:03 -0700
Subject: [PATCH] CMakeLists: Add repositories for system includes

Yocto's default aarch64 toolchain, i.e aarch64-poky-linux-, doesn't
include assert.h and stderr.h by default, leading to a compilation error
for files that need them.  As such add the path of the base sysroot
directory to the list of directories to be included when compiling.

Signed-off-by: Mathieu Poirier <mathieu.poirier@linaro.org>
---
 lib/arch/CMakeLists.txt | 1 +
 lib/libc/CMakeLists.txt | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/arch/CMakeLists.txt b/lib/arch/CMakeLists.txt
index 93c0ae76bb65..628f4c981051 100644
--- a/lib/arch/CMakeLists.txt
+++ b/lib/arch/CMakeLists.txt
@@ -12,6 +12,7 @@ target_link_libraries(rmm-lib-arch
 target_include_directories(rmm-lib-arch
     PUBLIC  "include"
             "include/${RMM_ARCH}"
+	    "../../../recipe-sysroot/usr/include/"
     PRIVATE "src/${RMM_ARCH}"
             "src/include")
 
diff --git a/lib/libc/CMakeLists.txt b/lib/libc/CMakeLists.txt
index 1631332dbc72..94d559b7aae0 100644
--- a/lib/libc/CMakeLists.txt
+++ b/lib/libc/CMakeLists.txt
@@ -12,7 +12,8 @@ if(NOT RMM_ARCH STREQUAL fake_host)
            rmm-lib-debug)
 
     target_include_directories(rmm-lib-libc SYSTEM
-        PUBLIC "include")
+        PUBLIC "include"
+        "../../../recipe-sysroot/usr/include/")
 
     target_sources(rmm-lib-libc
         PRIVATE "src/abort.c"
-- 
2.43.0

